### ISSUE TITLE
Syntax errors when building SDK demo app

### ISSUE THREAD

**Issue by hivemindunit** [aws-samples/aws-virtual-participant-framework-for-rtc#34]
### Expected behavior and actual behavior

- I walked through the steps described in the [deployment guide](https://github.com/aws-samples/aws-virtual-participant-framework-for-rtc/tree/main/virtual-participant-orchestrator-for-zoom-meeting#create-repository-and-secrets-for-cdk-deployment).
- Windows docker container was never created. I checked the CloudWatch logs, and found the following message in the stderr log:
`The command 'cmd /S /C call build_zoom_app.bat' returned a non-zero code: 1
An image does not exist locally with the tag: ************.dkr.ecr.us-east-1.amazonaws.com/zoom-virtual-participant-windows`
- I tried to build the app locally and multiple build errors fired:
Error	C2143	syntax error: missing '}' before 'constant'	sdk_demo_v2	C:\src\vcpkg\vcpkg\installed\x64-windows\include\smithy\tracing\TraceSpan.h	22	
Error	C2059	syntax error: 'constant'	sdk_demo_v2	C:\src\vcpkg\vcpkg\installed\x64-windows\include\smithy\tracing\TraceSpan.h	22	
and many more:
<img width="1920" alt="image" src="https://github.com/aws-samples/aws-virtual-participant-framework-for-rtc/assets/4390329/25e71e68-23f6-43e9-ab5f-12e305c80802">

### Steps to reproduce the problem

- Download the Zoom SDK Sample App
- Install dependencies with vcpgk: crow, aws-sdk-cpp[sns], cpp-jwt
- Apply the patch: https://github.com/aws-samples/aws-virtual-participant-framework-for-rtc/blob/main/virtual-participant-orchestrator-for-zoom-meeting/src/zoom_sdk_demo_v2.patch
- Switch to Release x64 configuration
- Build solution

### Specifications like the version of the project, operating system, or hardware

Windows 11, MS Visual Studio 2022 / 2019 Community Edition

**Comment by sinasojoodi** [comment]
It looks like a vcpkg dependency regression. Can you please verify the same error is popping up in your windows build logs captured in CloudWatch? you want to look at stdout for any build issues. the stderr output is expected (it's trying to load the image from ECR but can't since it was never built before).

> Inspect the windows build and container image generation build logs in the latest AWS CloudWatch log stream under the log group beggining with /aws/ssm/PipelineAppStage-CodePipelineCustomActionStack-RunBuildJobOnEc2Instance-

**Comment by hivemindunit** [comment]
Hey @sinasojoodi, thanks for the reply. 
As a quick and dirty fix to the above, I just added `#undef ERROR` at the top of TraceSpan.h source code. I ended up reinstalling VS, vcpkg and the packages, including aws-sdk-cpp.

You were right that I should have looked at the stdout logs. I don't recall what it was exactly, but something unrelated to the issue described above, so closing this one.

**Comment by sinasojoodi** [comment]
Thanks for letting me know is your issue resolved now?


On Sep 4, 2023, at 8:31 AM, Alex Tsimbalistov ***@***.***> wrote:

﻿

Hey @sinasojoodi<https://github.com/sinasojoodi>, thanks for the reply.
As a quick and dirty fix to the above, I just added #undef ERROR at the top of TraceSpan.h source code. I ended up reinstalling VS, vcpkg and the packages, including aws-sdk-cpp.

You were right that I should have looked at the stdout logs. I don't recall what it was exactly, but something unrelated to the issue described above, so closing this one.
